shell/src/main/java/alluxio/cli/bundler/command/CollectConfigCommand.java

```diff
@@ -52,10 +62,21 @@ public boolean hasSubCommand() {
   @Override
   public int run(CommandLine cl) throws AlluxioException, IOException {
     mWorkingDirPath = getWorkingDirectory(cl);
-    String confDir = mFsContext.getClusterConf().get(PropertyKey.CONF_DIR);
+    String confDirPath = mFsContext.getClusterConf().get(PropertyKey.CONF_DIR);
 
-    // TODO(jiacheng): phase 2 copy intelligently, check security risks
-    FileUtils.copyDirectory(new File(confDir), new File(mWorkingDirPath), true);
+    File confDir = new File(confDirPath);
+    List<File> allFiles = CommonUtils.recursiveListDir(confDir);
+    for (File f : allFiles) {
+      String relativePath = confDir.toURI().relativize(f.toURI()).getPath();
+      // Ignore file prefixes to exclude
+      for (String prefix : EXCLUDED_FILES) {
+        if (relativePath.startsWith(prefix)) {
```

Comment:
Is `relativePath` just the filename part?

Reply:
Your are right, when checking the prefix it should be checking against the filename. The `relativePath` is the path relative to the conf dir, when the config files can be in nested directories. I think we still need the `relativePath` when copying, as we want to preserve the directory structure.